ipn/ipnlocal/local.go

```diff
@@ -6456,7 +6456,7 @@ func suggestExitNode(report *netcheck.Report, netMap *netmap.NetworkMap, r *rand
 	var allowedCandidates set.Set[string]
 	if allowed, err := syspolicy.GetStringArray(syspolicy.AllowedSuggestedExitNodes, nil); err != nil {
 		return res, fmt.Errorf("unable to read %s policy: %w", syspolicy.AllowedSuggestedExitNodes, err)
-	} else if allowed != nil {
+	} else if allowed != nil && len(allowed) > 0 {
```

Comment:
@sailorfrag @clairew I wasn't entirely sure on this, but this logic seems reasonable (rather than allowing the entire set of exit nodes or adding another policy).  There appears to be another policy we enforce in the UI to hide this all together which covers that case.

Reply:
The bug here was that we were always sending back an empty result to the UI.

Reply:
As discussed on slack, `allowed != nil && len(allowed) == 0` is how a policy can state "do not permit suggestions". The empty result should signal to the UI that nothing meets the criteria for being a suggestion.